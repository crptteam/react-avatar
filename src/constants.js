export const SIZES = {
  small: 'small',
  large: 'large',
  medium: 'medium'
};

export const COLORS = {
  light: 'light',
  dark: 'dark'
};
