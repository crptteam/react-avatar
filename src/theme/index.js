import defaultTheme from './defaultTheme.js';

export { defaultTheme };
