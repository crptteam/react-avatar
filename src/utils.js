export function getAllPlainValuesAsObj(obj) {
  const plain = {};
  Object.keys(obj).forEach(
    key => (typeof obj[key] !== "object" ? (plain[key] = obj[key]) : null)
  );
  return plain;
}

export function getThemeAsPlainTextByKeys(theme, ...keys) {
  const plain = getAllPlainValuesAsObj(theme);

  keys.forEach(key => theme[key] && Object.assign(plain, theme[key]));

  return plain;
}

export function innerMerge(obj, ...others) {
  others.forEach(v => {
    for (const key in v) {
      if (typeof obj[key] === "object" && typeof v[key] === "object") {
        obj[key] = innerMerge({}, obj[key], v[key]);
      } else {
        obj[key] = v[key];
      }
    }
  });

  return obj;
}
