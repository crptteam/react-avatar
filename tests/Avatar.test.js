import React from "react";
import { Avatar } from "../src/";

describe("Avatar", () => {

  it("Should renders without problems", () => {

    const wrapper = shallow(
      <Avatar round abbr="ES" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});

