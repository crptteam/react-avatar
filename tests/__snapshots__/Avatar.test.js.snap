// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Should renders without problems 1`] = `
<AvatarWrap
  color="light"
  round={true}
  size="small"
  theme={
    Object {
      "dark": Object {
        "background": "#212C42",
        "color": "#FFFFFF",
      },
      "fontWeight": 400,
      "large": Object {
        "fontSize": "32px",
        "height": "128px",
        "width": "128px",
      },
      "light": Object {
        "background": "#FFFFFF",
        "color": "#212C42",
      },
      "small": Object {
        "fontSize": "14px",
        "height": "34px",
        "width": "34px",
      },
    }
  }
>
  <AvatarTextWrap
    color="light"
    size="small"
    theme={
      Object {
        "dark": Object {
          "background": "#212C42",
          "color": "#FFFFFF",
        },
        "fontWeight": 400,
        "large": Object {
          "fontSize": "32px",
          "height": "128px",
          "width": "128px",
        },
        "light": Object {
          "background": "#FFFFFF",
          "color": "#212C42",
        },
        "small": Object {
          "fontSize": "14px",
          "height": "34px",
          "width": "34px",
        },
      }
    }
  >
    ES
  </AvatarTextWrap>
</AvatarWrap>
`;
